src/coreclr/tools/Common/TypeSystem/Common/LayoutInt.cs

```diff
@@ -92,6 +92,20 @@ public string ToStringInvariant()
             return new LayoutInt(checked(left._value - right._value));
         }
 
+        public static LayoutInt AddThrowing(LayoutInt left, LayoutInt right, TypeDesc loadedType)
+        {
+            if (left.IsIndeterminate || right.IsIndeterminate)
+                return Indeterminate;
+
+            int result = left._value + right._value;
```

Comment:
Nit: My preferred way to do overflow checks is do the calculation with higher precision and then verify that there is no information lost by truncation to the original precision:
```
long result = (long)left._value + (long)right._value;
if ((int)result != result)
   throw
```
This is universal pattern that works for all operations.